### ISSUE TITLE
Issue with Loading Example Code

### ISSUE THREAD

**Issue by Petar-Luketina** [MichaelStott/KivMob#17]
Hey Michael,

First and foremost, thank you for taking the time to build this project, and creating the UI so visually appealing in the instructions. It gave me some ideas to improve my app's UI too!

As for the issue, I've been using all of your example code with no avail. It loads on my Android but then no ads appear. I used all the buildozer specs that you wrote. Furthermore, I'm really struggling to understand how to implement KivMob in my app, with multiple screens instead of the app screen.

My questions:
- Why isn't the code below displaying any ads?
(from https://github.com/MichaelStott/KivMob/blob/master/demo/main.py)
- How did everyone else learn to implement KivMob in their apps?

If possible, can someone share a recent project with buildozer specs that worked? Or a "Hello World" app they made to test KivMob. with buildozer specs.

Thank you in advance!!
      - Petar

```
from kivmob import KivMob
import kivy.utils
from kivy.app import App
from kivy.lang import Builder
from kivy.config import Config
from kivy.properties import ListProperty
from kivy.utils import platform
from kivy.uix.anchorlayout import AnchorLayout
from kivy.uix.floatlayout import FloatLayout
from kivy.uix.boxlayout import BoxLayout
from kivy.uix.button import Button
from kivy.uix.behaviors import ButtonBehavior
from kivy.uix.image import Image
from kivy.uix.label import Label
from kivy.uix.popup import Popup
from kivy.core.window import Window

if platform not in ('android', 'ios'):
    # Approximate dimensions of mobile phone.
    Config.set('graphics', 'resizable', '0')
    Window.size = (320, 420)
import webbrowser

__version__ = "1.0"

Builder.load_string("""
#:import kivy kivy
<KivMobDemoUI>:
    canvas.before:
        Color:
            rgba: kivy.utils.get_color_from_hex("f5f5f5")
        Rectangle:
            pos: self.pos
            size: self.size
    FloatLayout:
        Image:
            source: "assets/kivmob-title.png"
    BoxLayout:
        orientation: "vertical"
        AnchorLayout:
            CustomButton:
                text: "Toggle Banner"
                size_hint: 0.8, 0.2
                on_release: app.toggle_banner()
        AnchorLayout:
            CustomButton:
                text: "Show Interstitial"
                size_hint: 0.8, 0.2
                on_release: app.show_interstitial()
<CustomButton>:
    canvas.before:
        Color:
            rgba: self.box_color
        RoundedRectangle:
            pos: self.pos
            size: self.size
    color: (1, 1, 1, 1)
    bold: True
    text_size: self.width, None
    height: self.texture_size[1]+size_hint_y: None
    text_size: self.width, None
    height: self.texture_size[1] + sp(20)
    halign: 'center'
    valign: 'middle'
""")


class KivMobDemoUI(FloatLayout):
    pass


class CustomButton(ButtonBehavior, Label):
    """Custom button used by applicatiod rather than default kivy
    button.

    Attributes:
        box_color -- Button color.
        prv_color -- Stores original box color.
    """

    box_color = ListProperty(kivy.utils.get_color_from_hex("56b669"))
    prv_color = ListProperty(kivy.utils.get_color_from_hex("56b669"))

    def __init__(self, **kwargs):
        super(CustomButton, self).__init__(**kwargs)
        self.always_release = True

    def on_press(self):
        self.prv_color = self.box_color
        self.box_color = (self.box_color[0] * 0.5, \
                          self.box_color[1] * 0.5, \
                          self.box_color[2] * 0.5, \
                          self.box_color[3])

    def on_release(self):
        self.box_color = self.prv_color


class KivMobDemo(App):


    def build(self):
        APP_ID = "ca-app-pub-"
        BANNER_ID = "ca-app-pub-"
        INTERSTITIAL_ID = "ca-app-pub-"
        TEST_DEVICE_ID = "ca-app-pub-"

        self.ads = KivMob(APP_ID)
        self.ads.add_test_device(TEST_DEVICE_ID)
        self.ads.new_banner({"unitID": BANNER_ID})
        self.ads.new_interstitial(INTERSTITIAL_ID)
        self.ads.request_banner()
        self.ads.show_banner()
        self.ads.request_interstitial()
        self.toggled = False
        return KivMobDemoUI()

    def on_start(self):
        """Called on application start.
        """
        if platform not in ("android", "ios"):
            self.desktop_warning()

    def desktop_warning(self):
        layout = BoxLayout(orientation='vertical')
        layout.add_widget(Label(text='KivMob will not display ads on ' + \
                                     'nonmobile platforms. You must build an ' + \
                                     'Android project to demo ads. (iOS not yet ' + \
                                     'supported)',
                                size_hint_y=1,
                                text_size=(250, None),
                                halign='left',
                                valign='middle'))
        button_layout = BoxLayout()
        button1 = Button(text="Open Build Steps", size_hint=(0.8, 0.2))
        button1.bind(on_release=lambda x:
        webbrowser.open("https://www.google.com"))
        button_layout.add_widget(button1)
        button2 = Button(text="Close", size_hint=(0.8, 0.2))
        button2.bind(on_release=lambda x: popup.dismiss())
        button_layout.add_widget(button2)
        layout.add_widget(button_layout)
        popup = Popup(title='KivMob Demo Alert',
                      content=layout,
                      size_hint=(0.9, 0.9))
        popup.open()

    def interstitial_warning(self):
        layout = BoxLayout(orientation='vertical')
        layout.add_widget(Label(text="Ad has not loaded. " + \
                                     "Wait a few seconds and then " + \
                                     "try again.",
                                size_hint_y=1,
                                text_size=(250, None),
                                halign='left',
                                valign='middle'))
        button_layout = BoxLayout()
        close = Button(text="Close", size_hint=(0.8, 0.2))
        close.bind(on_release=lambda x: popup.dismiss())
        button_layout.add_widget(close)
        layout.add_widget(button_layout)
        popup = Popup(title='KivMob Demo Alert',
                      content=layout,
                      size_hint=(0.9, 0.9))
        popup.open()

    def on_pause(self):
        """Android specific method. Save important app data on pause.
        """
        return True

    def on_resume(self):
        """Android specific method. Resume paused app.
        """
        self.ads.request_interstitial()

    def toggle_banner(self):
        if not self.toggled:
            self.ads.show_banner()
        else:
            self.ads.hide_banner()
        self.toggled = not self.toggled

    def show_interstitial(self):
        if self.ads.is_interstitial_loaded():
            self.ads.show_interstitial()
        else:
            self.interstitial_warning()


if __name__ == "__main__":
    KivMobDemo().run()
```

**Comment by MichaelStott** [comment]
Edited your post to remove App/Banner/Interstitial Ids.

Hard to say without the logcat output. There can be a delay between setting up Admob and beginning to receive/serve ads. I'd wait a while before trying again.

**Comment by thegameguy** [comment]
Hi Petar, as you say the ads appear once on your phone and do not disappear again. You are may be requesting the ad only once. You need to request an ad before everytime you show an ad. Hope it helps

**Comment by Petar-Luketina** [comment]
@MichaelStott , thank you so much for the quick response and editing my post! I'll wait a couple more days, then try it again with logcat.

@thegameguy thank you too for trying to help. But I was saying that the buttons appear, and not the ads. Hopefully the ads will appear in a couple days. If not, I'll post my logcat if I can't figure out the problem myself.

Thanks again you two!

**Comment by thegameguy** [comment]
Hey man, if you ran the code at is that you mention, just making sure you have followed these steps.
- You made an account with Google Admob.
- You acquired the App id from your account and set it in your app where it says App ID.
- You have set the sample ad units to see sample Banner and Interstitial ads. links for sample ad units
https://developers.google.com/admob/android/test-ads#sample_ad_units
- You have set test device for initial testing.

**Comment by Petar-Luketina** [comment]
Hey guys. Wonderful news! It finally worked, just like you said! Google emailed me telling me to finish my registration. Apparently my payment method didn't go through the first time.

Thanks again @thegameguy and especially @MichaelStott for making KivMob!

**Comment by solari985** [comment]
Hi All, I have similar issues what were mentioned before (in issue 5 and in this). My test interstitial ad is shown only after tapping the button 2 times (from test code) and when I try without button, it doesn't appear at all, self.ads.is_interstitial_loaded() = False and nothing happens, also nothing at 2nd time after request and show again. Why this happens? Why the button needed? It is hard to imagine that it is related to the account and the payment, but could be that the case? Thank you in advance.

**Comment by akgupta0777** [comment]
@solari985 
The interstitial ad is shown on **2 taps** because on the **first tap** on button It requests an Ad and loads the interstitial in background. It takes about 5 seconds to load the ad. After you tap on the button for the **second time** It finally shows you the interstial Ad.

Interstitial Ads are meant to be loaded in background before and when the time comes you show the ad by calling show_interstitial() method.

I Will further implement the interstitial ad without button and post the code if it works.

Hope it helps.

**Comment by solari985** [comment]
@akgupta0777 
First I tried the ad to be shown at exit, that didn't work as I described here and in issue 64, but in the meantime I could manage the ad to appear without button with inserting some delay (sometimes appeared, sometimes not), but with the info that it needs 5 seconds to be loaded I could manage it in a proper way. Since then I haven't tried the ad to be shown at exit, but now it is not neccessary, it is shown between screens. Loaded earlier and shown later. Thank you for the 5 seconds info.
